src/app/profile/TransactionHistory.tsx

```diff
@@ -0,0 +1,131 @@
+
+  return (
+    <div className="my-2 px-4">
+      <p className="py-4 text-3xl font-bold">Transaction History</p>
+      {isLoading ? (
+        <div data-testid="loader" className="loader"></div>
+      ) : (
+        <div className="overflow-x-scroll w-screen md:w-11/12 px-4 lg:px-0">
+          <table className="w-full border-collapse table rounded-lg overflow-hidden border-emerald-500">
+            <thead className="table-header-group sticky top-0">
+              <tr className="border table-row text-xl">
+                <th className="p-2 border-2 border-emerald-500 text-left table-cell">
+                  Checkout Time
+                </th>
+                <th className="p-2 border-2 border-emerald-500 text-left table-cell">
+                  Package Plan
+                </th>
+                <th className="p-2 border-2 border-emerald-500 text-left table-cell">
+                  Price
+                </th>
+                <th className="p-2 border-2 border-emerald-500 text-left table-cell">
+                  Expiry Time
+                </th>
+                <th className="p-2 border-2 border-emerald-500 text-left table-cell">
+                  Status
+                </th>
+              </tr>
+            </thead>
+            <tbody className="table-row-group">
+              {data && data?.length > 0 ? (
+                data
+                  ?.filter((transaction) => {
+                    if (status) return transaction.status === status
+                    return transaction
+                  })
+                  .map(
+                    ({
+                      checkout_time,
+                      expiry_time,
+                      package: { name },
+                      price,
+                      id,
+                      status,
+                      midtrans_url,
+                    }) => (
+                      <tr
+                        key={id}
+                        className={twMerge(
+                          'border table-row text-lg group',
+                          HoverRowVariants({
+                            hover: getTransactionStatus(
+                              status as TransactionStatus
+                            ),
+                          })
+                        )}
+                      >
+                        <td className="p-2 border-2 border-emerald-500">
+                          {dayjs(checkout_time).format('ddd, D MMM YY HH:mm')}
+                        </td>
+
+                        <td className="p-2 border-2 border-emerald-500">
+                          {name}
+                        </td>
+                        <td className="p-2 border-2 border-emerald-500">
+                          {formatRupiah(price)}
+                        </td>
+                        <td className="p-2 border-2 border-emerald-500">
+                          {dayjs(expiry_time).format('ddd, D MMM YY HH:mm')}
```

Comment:
I think there are plenty of time in our projects that we use dayjs and format it to a certain format. I think it's better to make utils function for better code readability.

Reply:
ok thanks for suggesting this, I have rafactored the code and make it more modular.